src/routes/transactions/mappers/common/human-description.mapper.ts

```diff
@@ -0,0 +1,128 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { formatUnits, isAddress, isHex } from 'viem';
+import { ITokenRepository } from '../../../../domain/tokens/token.repository.interface';
+import { TokenRepository } from '../../../../domain/tokens/token.repository';
+import { Token } from '../../../../domain/tokens/entities/token.entity';
+import { MAX_UINT256 } from '../../constants';
+import {
+  ILoggingService,
+  LoggingService,
+} from '../../../../logging/logging.interface';
+import { IHumanDescriptionRepository } from '../../../../domain/human-description/human-description.repository.interface';
+import { HumanDescriptionRepository } from '../../../../domain/human-description/human-description.repository';
+import {
+  HumanDescriptionFragment,
+  ValueType,
+} from '../../../../domain/human-description/entities/human-description.entity';
+import { MultisigTransaction } from '../../../../domain/safe/entities/multisig-transaction.entity';
+import { ModuleTransaction } from '../../../../domain/safe/entities/module-transaction.entity';
+import { isMultisigTransaction } from '../../../../domain/safe/entities/transaction.entity';
+import { SafeAppInfoMapper } from './safe-app-info.mapper';
+
+@Injectable()
+export class HumanDescriptionMapper {
+  private static SIG_HASH_INDEX = 10;
+
+  constructor(
+    @Inject(ITokenRepository) private readonly tokenRepository: TokenRepository,
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(IHumanDescriptionRepository)
+    private readonly humanDescriptionRepository: HumanDescriptionRepository,
+    private readonly safeAppInfoMapper: SafeAppInfoMapper,
+  ) {}
+
+  async mapHumanDescription(
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<string | null> {
+    if (!transaction.data || !isHex(transaction.data) || !transaction.to) {
+      return null;
+    }
+
+    const dataStart = transaction.data.slice(
+      0,
+      HumanDescriptionMapper.SIG_HASH_INDEX,
+    );
+    const sigHash = isHex(dataStart) ? dataStart : null;
+    if (!sigHash) return null;
+
+    let token: Token | null = null;
+    try {
+      token = await this.tokenRepository.getToken({
+        chainId,
+        address: transaction.to,
+      });
+    } catch (error) {
+      this.loggingService.debug(`Error trying to get token: ${error.message}`);
+    }
+
+    try {
+      const descriptionFragments =
+        this.humanDescriptionRepository.getHumanDescription({
+          functionSignature: sigHash,
+          to: transaction.to,
+          data: transaction.data,
+        });
+
+      const description = this.createHumanDescription(
+        descriptionFragments,
+        token,
+      );
+
+      const safeAppInfo = isMultisigTransaction(transaction)
+        ? await this.safeAppInfoMapper.mapSafeAppInfo(chainId, transaction)
+        : null;
+
+      return safeAppInfo
+        ? `${description} via ${safeAppInfo.name}`
```

Comment:
Something to consider for the future: should this be responsibility of the domain layer or the route?

Right now we are including the Safe App Info as part of the description whenever we have it. One can see this as an extension to every template that we have listed and can probably be formatted in a different way (e.g.: to include the safe app logo).